lib/graphql_devise/rails/routes.rb

```diff
@@ -1,111 +1,98 @@
 module ActionDispatch::Routing
   class Mapper
+    DEVISE_OPERATIONS = [
+      :sessions,
+      :registrations,
+      :passwords,
+      :confirmations,
+      :omniauth_callbacks,
+      :unlocks,
+      :invitations
+    ].freeze
+
     def mount_graphql_devise_for(resource, opts = {})
       custom_operations    = opts.fetch(:operations, {})
       skipped_operations   = opts.fetch(:skip, [])
       only_operations      = opts.fetch(:only, [])
       additional_mutations = opts.fetch(:additional_mutations, {})
       additional_queries   = opts.fetch(:additional_queries, {})
-
-      if [skipped_operations, only_operations].all?(&:any?)
-        raise GraphqlDevise::Error, "Can't specify both `skip` and `only` options when mounting the route."
-      end
-
-      default_mutations = {
-        login:               GraphqlDevise::Mutations::Login,
-        logout:              GraphqlDevise::Mutations::Logout,
-        sign_up:             GraphqlDevise::Mutations::SignUp,
-        update_password:     GraphqlDevise::Mutations::UpdatePassword,
-        send_password_reset: GraphqlDevise::Mutations::SendPasswordReset,
-        resend_confirmation: GraphqlDevise::Mutations::ResendConfirmation
-      }.freeze
-      default_queries = {
-        confirm_account:      GraphqlDevise::Resolvers::ConfirmAccount,
-        check_password_token: GraphqlDevise::Resolvers::CheckPasswordToken
+      path                 = opts.fetch(:at, '/graphql_auth')
+      mapping_name         = resource.underscore.tr('/', '_').to_sym
+      authenticatable_type = opts[:authenticatable_type].presence ||
+        "Types::#{resource}Type".safe_constantize ||
+        GraphqlDevise::Types::AuthenticatableType
+      param_operations     = {
+        custom:  custom_operations,
+        only:    only_operations,
+        skipped: skipped_operations
       }
-      supported_operations = default_mutations.keys + default_queries.keys
 
-      unless skipped_operations.all? { |skipped| supported_operations.include?(skipped) }
-        raise GraphqlDevise::Error, 'Trying to skip a non supported operation. Check for typos.'
-      end
-      unless only_operations.all? { |only| supported_operations.include?(only) }
-        raise GraphqlDevise::Error, 'One of the `only` operations is not supported. Check for typos.'
-      end
-
-      path         = opts.fetch(:at, '/graphql_auth')
-      mapping_name = resource.underscore.tr('/', '_').to_sym
+      validate_gql_devise_operations!(param_operations)
```

Comment:
Still think you should call the new class method here directly instead of adding another method to rails router